lib/msf/core/exploit/ldap.rb

```diff
@@ -7,83 +7,160 @@
 require 'net-ldap'
 
 module Msf
-module Exploit::Remote::LDAP
-
-  def initialize(info = {})
-    super
+  module Exploit::Remote::LDAP
+    def initialize(info = {})
+      super
+
+      register_options([
+        Opt::RHOST,
+        Opt::RPORT(389),
+        OptBool.new('SSL', [false, 'Enable SSL on the LDAP connection', false]),
+        OptString.new('BIND_DN', [false, 'The username to authenticate to LDAP server']),
+        OptString.new('BIND_PW', [false, 'Password for the BIND_DN'])
+      ])
+
+      register_advanced_options([
+        OptFloat.new('LDAP::ConnectTimeout', [true, 'Timeout for LDAP connect', 10.0])
+      ])
+    end
 
-    register_options([
-      Opt::RHOST,
-      Opt::RPORT(389),
-      OptBool.new('SSL', [false, 'Enable SSL on the LDAP connection', false])
-    ])
+    def rhost
+      datastore['RHOST']
+    end
 
-    register_advanced_options([
-      OptFloat.new('ConnectTimeout', [true, 'Timeout for LDAP connect', 10.0])
-    ])
-  end
+    def rport
+      datastore['RPORT']
+    end
 
-  def rhost
-    datastore['RHOST']
-  end
+    def peer
+      "#{rhost}:#{rport}"
+    end
 
-  def rport
-    datastore['RPORT']
-  end
+    def ldap_connect(opts = {}, &block)
+      connect_opts = {
+        host: rhost,
+        port: rport,
+        connect_timeout: datastore['ConnectTimeout']
```

Comment:
This and any other use should have been updated to `LDAP::ConnectTimeout`.

Reply:
corrected